data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/file/DynamicFileReader.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.file;
+
+import java.io.InputStream;
+
+public interface DynamicFileReader {
+    InputStream getFile();
```

Comment:
I tend to think that this should be named `readFile`. Also, it should probably take in a String for the filename.

```
InputStream readFile(String filename);
```